app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala

```diff
@@ -40,40 +41,39 @@ class QueueController @Inject()(queueService: QueueService,
                                 fieldsService: ApiSubscriptionFieldsService,
                                 idGenerator: NotificationIdGenerator,
                                 dateTimeProvider: DateTimeProvider,
-                                cdsLogger: CdsLogger) extends BaseController {
+                                logger: NotificationLogger) extends BaseController {
 
-  private val SUBSCRIPTION_FIELD_HEADER_NAME = "api-subscription-fields-id"
-  private val CLIENT_ID_HEADER_NAME = "X-Client-ID"
-
-  private val MISSING_CLIENT_ID_ERROR = s"$CLIENT_ID_HEADER_NAME required"
+  private val MISSING_CLIENT_ID_ERROR = s"$X_CLIENT_ID_HEADER_NAME required"
   private val MISSING_BODY_ERROR = "Body required."
 
-  def save(): Action[AnyContent] = Action.async {
-    implicit request => {
-      val headers = request.headers
-      val conversationIdMsg = headers.get("X-Conversation-ID").fold(s"[conversationId not found] Headers=$headers"){ cid => s"[conversationId=$cid]"}
-      cdsLogger.debug(s"saving request - $conversationIdMsg")
-      getClientId(headers, conversationIdMsg).flatMap(_.fold(Future.successful(BadRequest(MISSING_CLIENT_ID_ERROR))) {
-        clientId =>
-          request.body.asXml.fold(Future.successful(BadRequest(MISSING_BODY_ERROR))) { body =>
-            queueService.save(
-              clientId,
-              Notification(
-                idGenerator.generateId(),
-                headers.remove(CLIENT_ID_HEADER_NAME, SUBSCRIPTION_FIELD_HEADER_NAME).toSimpleMap,
-                body.toString(),
-                dateTimeProvider.now(),
-                None
-              )
-            ).map(notification =>
-              Result(ResponseHeader(CREATED, Map(LOCATION -> routes.QueueController.get(notification.notificationId).url)), HttpEntity.NoEntity))
-          }
-
-      })
-    }
+  def save(): Action[AnyContent] = Action.async { implicit request =>
+    val headers = request.headers
+    logger.info(s"saving request", headers.headers)
```

Comment:
we have a mixture of concerns in `getClientId` 
1) validate CLIENT_ID is present 
2) lookup SUBSCRIPTION_FIELDS)ID and validate that it is a UUID.
I suggest doing this in 2 steps. You could reuse the `validateHeader(headers: Headers, endpointName: String): Either[ErrorResponse, String]` to help to do this.

Reply:
Since this was a logging PR i didn't unpick too much existing code. I'll have a look at this though